dot/node.go

```diff
@@ -146,29 +146,27 @@ func NodeInitialized(basepath string) bool {
 }
 
 // LoadGlobalNodeName returns the stored global node name from database
-func LoadGlobalNodeName(basepath string) (string, error) {
+func LoadGlobalNodeName(basepath string) (nodename string, err error) {
 	// initialise database using data directory
 	db, err := utils.SetupDatabase(basepath, false)
 	if err != nil {
 		return "", err
 	}
 
 	defer func() {
-		err = db.Close()
-		if err != nil {
-			logger.Errorf("failed to close database: %s", err)
+		e := db.Close()
```

Comment:
nit, maybe instead of logging the close error:

1. Rename `e` to `closeErr` (sorry I'm more of an entire words person that single letter C style names 😄) 
2. If `err` (the returned error) is nil, set it to `closeErr`

Reply:
updated.